### PR TITLE
App maintenance [Sprint 8]

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- [x] Add checks for internet connection before refreshing.
- [x] Logs account deletion.
- [x] Reduces splash screen load time.
- [x] Fixes bug on insights hourly graph.
- [x] updates android and ios versions
- [x] Removes build number from ap version.
- [x] Fixes bug on datetime display.
- [x] Subsistutes CircularProgressIndicator  with CupertinoActivityIndicator

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Follow instructions in the [Readme file](https://github.com/airqo-platform/AirQo-frontend/tree/staging/mobile#get-started)

#### What are the relevant tickets?
- [MOB-70](https://airqoteam.atlassian.net/browse/MOB-70)

### CODE REVIEW

File: mobile/lib/screens/about_page.dart [hunk 1]
```diff
@@ -52,7 +52,7 @@ class _AboutAirQoState extends State<AboutAirQo> {
                     height: 12,
                   ),
                   Text(
-                    '${_packageInfo.version}(${_packageInfo.buildNumber})',
```
Comment (NoahNsimbe) [inline mobile/lib/screens/about_page.dart:55]: removing build number from app version

File: mobile/lib/utils/data_formatter.dart [hunk 2]
```diff
@@ -44,7 +43,7 @@ List<List<charts.Series<Insights, String>>> insightsChartData(
 
       insightsGraphs.add([
         charts.Series<Insights, String>(
-          id: 'Insights ${pollutant.toTitleCase()} Chart data',
+          id: '${const Uuid().v4()}-${earliestDate.day}',
```
Comment (NoahNsimbe) [inline mobile/lib/utils/data_formatter.dart:46]: using a unique id for every chart

File: mobile/lib/widgets/custom_shimmer.dart [hunk 3]
```diff
@@ -40,11 +41,11 @@ void loadingScreen(BuildContext _context) async {
       barrierDismissible: false,
       builder: (ctx) => Container(
           decoration:
-              BoxDecoration(color: Config.appColorBlue.withOpacity(0.05)),
-          child: Center(
-              child: CircularProgressIndicator(
+              BoxDecoration(color: Config.appColorBlue.withOpacity(0.005)),
+          child: CupertinoActivityIndicator(
+            radius: 20,
             color: Config.appColorBlue,
```
Comment (NoahNsimbe) [inline mobile/lib/widgets/custom_shimmer.dart:47]: Subsitututing CircularProgressIndicator with CupertinoActivityIndicator

File: mobile/pubspec.yaml [hunk 4]
```diff
@@ -14,7 +14,7 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 # Read more about iOS versioning at
 # https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
 # android test
-version: 2.0.5+20011
+version: 2.0.6+20012
```
Comment (NoahNsimbe) [inline mobile/pubspec.yaml:17]: updating android version

File: mobile/ios/Runner.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -392,7 +392,7 @@
 					"$(inherited)",
 					"@executable_path/Frameworks",
 				);
-				MARKETING_VERSION = 2.0.6;
+				MARKETING_VERSION = 2.0.7;
```
Comment (NoahNsimbe) [inline mobile/ios/Runner.xcodeproj/project.pbxproj:395]: updating iOS version

File: mobile/lib/screens/search_page.dart [hunk 6]
```diff
@@ -496,7 +496,7 @@ class _SearchPageState extends State<SearchPage> {
             ),
         enableSuggestions: true,
         cursorWidth: 1,
-        autofocus: true,
+        autofocus: false,
```
Comment (NoahNsimbe) [inline mobile/lib/screens/search_page.dart:499]: removing autofocus from search page

File: mobile/pubspec.yaml (hunk 4 above)
Comment (Baalmart) [reply mobile/pubspec.yaml:17]: just curious about this manual update of the android versions. Could we explore just stating the minimum (starting) version or an approach that does not involve the manual update of versions.

**Review by Baalmart — APPROVED** [review]

**Review by 123MwanjeMike — APPROVED** [review]
LGTM

File: mobile/pubspec.yaml (hunk 4 above)
Comment (NoahNsimbe) [reply mobile/pubspec.yaml:17]: unfrortunately play store and app store dont support auto updating. It is the developer to set the version or have it configured in CI/CD @Baalmart

**Pull request merged**